### PR TITLE
Bug 1149164 - Add support for ingesting Android build file sizes (or other arbitrary data) in Perfherder

### PR DESCRIPTION
**Pull request by wlach**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1094)

### CODE REVIEW

File: treeherder/log_parser/parsers.py [hunk 1]
```diff
@@ -465,3 +465,26 @@ def parse_line(self, line, lineno):
...
+
+    def __init__(self):
+        super(PerformanceParser, self).__init__("performance_data")
+
+    def parse_line(self, line, lineno):
+        match = self.RE_PERFORMANCE.match(line)
+        if match:
+            # this will throw an exception if the json parsing breaks, but
+            # that's the behaviour we want
+            self.artifact = json.loads(match.group(1))
```
Comment (jmaher) [inline treeherder/log_parser/parsers.py]: what if json.loads fails, what is the error handling path?

File: treeherder/etl/perf.py [hunk 2]
```diff
@@ -67,7 +67,8 @@ def _get_signature_hash(signature_properties):
 
 
 def load_perf_artifacts(project_name, reference_data, job_data, datum):
-    perf_datum = json.loads(datum['blob'])
+    blob = json.loads(datum['blob'])
+    perf_datum = blob['performance_data']
```
Comment (jmaher) [inline treeherder/etl/perf.py]: this spot is failing in the tests, do we always assume 'performance_data' exists in the blob?

File: treeherder/log_parser/parsers.py (hunk 1 above)
Comment (wlach) [inline treeherder/log_parser/parsers.py]: The log parser will throw an exception and we'll consider the job to have failed.

File: treeherder/etl/perf.py (hunk 2 above)
Comment (wlach) [inline treeherder/etl/perf.py]: Yeah, we do assume this. If this happened in the field, we'd return a 500 error.

This reminds me that autophone will need some small updates to support the fact that the performance artifact is now wrapped in a dictionary (its key being performance_data). I can provide a patch.

Comment (wlach) [inline treeherder/etl/perf.py]: Update: tests should be fixed now.

**Comment by edmorley** [comment]
The change to tests/e2e/test_perf_ingestion.py needs to be in the first commit rather than the second yeah? Other than that lgtm :-)

**Pull request merged**